docs/devel/api_changes.md

```diff
@@ -0,0 +1,237 @@
+# So you want to change the API?
+
+The Kubernetes API has two major components - the internal API structures and
+the versioned API structures.  The versioned APIs are intended to be stable,
+while the internal API is implemented to best reflect the needs of the
```

Comment:
I'm fine with that, but they do live in pkg/api and they are the rough parallel of the versioned APIs.  Will make this change.